server/backend/news.go

```diff
@@ -34,3 +36,39 @@ func (s *CampusServer) GetNewsSources(ctx context.Context, _ *emptypb.Empty) (ne
 	}
 	return &pb.NewsSourceReply{Sources: resp}, nil
 }
+
+func (s *CampusServer) GetNews(ctx context.Context, req *pb.GetNewsRequest) (newsSources *pb.GetNewsReply, err error) {
+	if err = s.checkDevice(ctx); err != nil {
+		return
+	}
+
+	var sources []model.News
```

Comment:
These aren't "news sources" but actual news entries, no? Seems the name for the variable is a bit off.